123_state_tasks.R

```diff
@@ -17,19 +17,40 @@ do_state_tasks <- function(oldest_active_sites, ...) {
     }
   )
 
+  plot_step <- create_task_step(
+    step_name = 'plot',
+    target_name = function(task_name, ...) {
+      sprintf("3_visualize/out/timeseries_%s.png", task_name)
+    },
+    command = function(steps, ...) {
+      sprintf("plot_site_data(target_name, %s, parameter)",
+              steps[["download"]]$target_name)
+    }
+  )
+
+  tally_step <- create_task_step(
+    step_name = "tally",
+    target_name = function(task_name, ...) {
+      sprintf("%s_tally", task_name)
+    },
+    command = function(steps, ...) {
+      sprintf("tally_site_obs(%s)", steps[["download"]]$target_name)
+    }
+  )
+
   # Create the task plan
   task_plan <- create_task_plan(
     task_names = state_abbv,
-    task_steps = list(download_step),
+    task_steps = list(download_step, plot_step, tally_step),
     add_complete = FALSE)
 
   # Create the task remakefile
   create_task_makefile(
     task_plan = task_plan,
     makefile = '123_state_tasks.yml',
     include = 'remake.yml',
-    sources = '1_fetch/src/get_site_data.R',
-    packages = c('tidyverse', 'dataRetrieval'),
+    sources = c(...),
```

Comment:
Do you feel clearer on the distinction between function arguments and `depends` items in the `state_tasks` target now?

Reply:
(I did add an issue to ds-pipelines-3 about clarifying `depends` for posterity in that earlier section)

Reply:
The function arguments are being used downstream to serve a second purpose, so now the difference between passing them in feels like `...` are for files with needed R fxns and `depends:` are for objects. Is there now way to pass a named argument that is a list of the source files instead to be clearer? E.g., 

```

Reply:
I would prefer it to be a named argument, too, but I _think_ what you're proposing above wouldn't let scipiper identify those source files as dependencies. But I haven't actually tested...if that _does_ work, I'd much prefer your solution! Would you be up for experimenting with it while you have this simple example in hand? Does scipiper allow you to define the arguments in that way (without `I()` around the `c()`)? If so, then if you edit `get_site_data.R`, does scipiper detect the need to rebuild?

Reply:
I don’t think it works because you can’t do a function within a function for a recipe, and c() is a function. Would be good to know if it works though!

Reply:
Ah right, yeah that doesn't work. It doesn't understand `c()` within a function in the `yml`. You could do:

```
# Pull site data
  state_tasks_sources:
    command: c(
      '1_fetch/src/get_site_data.R', 
      '2_process/src/tally_site_obs.R',
      '3_visualize/src/plot_site_data.R')
  state_tasks:
    command: do_state_tasks(
      oldest_active_sites,
      source_files = state_tasks_sources)
    depends:
      - parameter
```

but that is creating another target in the top-level yaml and doesn't really simplify things.

Reply:
Thanks for trying it out, and oh well. Your most recent suggestion would rebuild `state_tasks_sources` when the code files change but wouldn't rebuild `state_tasks` because the contents of `state_tasks_sources` wouldn't change. So I think we're stuck with `...` for now.

Reply:
`...` ain't so bad...
(says someone who probably overuses it! 😆 )

Reply:
Ah yes, darn

Reply:
I agree with Lindsay that named arguments are clearer when they're an option, and I dislike that you only get one `...` to work with per function (unlike regular arguments), but I definitely use `...` a lot in my prose...